src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -142,12 +144,17 @@ public Task StartAsync(ITransport transport, HttpClient httpClient)
             {
                 Interlocked.Exchange(ref _connectionState, ConnectionState.Disconnected);
 
-                // Do not "simplify" - events can be removed from a different thread
-                var closedEventHandler = Closed;
-                if (closedEventHandler != null)
-                {
-                    closedEventHandler(t.IsFaulted ? t.Exception.InnerException : null);
-                }
+                _eventQueue.Enqueue(queue => {
```

Comment:
This is the one that had `ignore2` but since we are in a non-async function (lambda) here the compiler does not complain about not assigning the returned task so I did remove it.